### DISCUSSION TITLE
[privacy/security enhancement]: Maximum isolated cli for Docker

### DISCUSSION THREAD

**Discussion by gwpl** [open-webui/open-webui#8150] [General]
As community had discussion with statements like: "This is fairly concerning. This project has the potential to allow people to actually have privacy while using AI but some of the design choices do not seem to focus on privacy." 
or "Don't trust our word for it; We encourage you to examine/audit our code",
considering that auditing whole codebase it not economically feasible for most users,
here https://github.com/open-webui/open-webui/discussions/1966#discussioncomment-11676778
I linked to this proposal:

Could you provide commandline (in some e.g. bash script file or some README), that runs things in docker with maxium isolation! What does it mean?

Topic of isolating programs running in Docker for security purposes, is explored here:

* https://github.com/jessfraz/dotfiles/blob/master/.dockerfunc - bash functions wrapping calls
* https://github.com/jessfraz/dockerfiles - docker files for above file
* https://youtu.be/cYsVvV1aVss - lecture by Jess Fraz about using "Docker as a Desktop" to run everything in Dockers with increased isolation.

It basically means to set all isolation parameters as strict as possible, and allow only the most necessary minium.

Additionally one could consider additional even more enhanced setup where all traffic goes through firewall/gateway running in another docker container (e.g. both can be in same docker network and docker-compose?), so all external connections to whatever destinations would have to go thought approved whitelist and be audited/logged (?).

**Comment by gaby** [comment]
I don't think this is Open-WebUI job to implement. They are already limited given a lot of these tools require root to access the GPU's and resources.

The Docker image allows overriding the UID/GID but it's not tested.

See here: https://github.com/open-webui/open-webui/blob/main/Dockerfile#L19

Regarding firewall, that's the job of the dev/team using Open-WebUI to control access to their instance.

**Reply by gwpl** [reply]
> Regarding firewall, that's the job of the dev/team using Open-WebUI to control access to their instance.

I apologise , because open-webui was introduced to me by enthusiasts hobbysts running it on their own machine I thought it would be nice courtesy to such home users to provide more isolated setup,

instead I read in other ticket I quoted earlier someone was writing people to audit open-webui code on their own...

It sounds like you are targeting more enterprise userbase with such capacities and resources.

> They are already limited given a lot of these tools require root to access the GPU's and resources.

How is it related? 

Open web-ui is just front end,
and how compute is performed via api e.g. by ollama or some provider is isolated to backend... which may not even run on their machine...

**Reply by gwpl** [reply]
So I just thought about providing proposal what else could users community look for, additionally to "audit it yourself" advice...

I am not expecting maintainers who alredy have a lot of work to do it... actually if this work would be even provided in separate repo "open-webui-isolated" and linked from documentation, still good... 

I am just sharing with community alternative to answer that I consider unfeasible, I don't expect home users or small businesses to have budget to conduct security audits...
